src/js/navIcons.js

```diff
@@ -0,0 +1,127 @@
+/*
+
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.firstDiscovery");
+
+    /*
+     * The nav icon
+     */
+    fluid.defaults("gpii.firstDiscovery.icon", {
+        gradeNames: ["fluid.viewComponent", "autoInit"],
+        position: null,  // must be supplied by integrators
+        selectors: {
+            activeIndicator: ".gpiic-fd-activeIndicator",
+            doneIndicator: ".gpiic-fd-doneIndicator"
+        },
+        styles: {
+            active: "gpii-fd-active",
+            show: "gpii-fd-show"
+        },
+        modelListeners: {
+            "isActive.setActiveState": {
+                listener: "{that}.setActiveState",
+                args: ["{change}.value"]
+            },
+            "isVisited.setVisitedState": {
+                listener: "{that}.setVisitedState",
+                args: ["{change}.value"]
+            }
+        },
+        invokers: {
+            setActiveState: {
+                funcName: "gpii.firstDiscovery.icon.setActiveState",
+                args: ["{that}", "{arguments}.0"]
+            },
+            setVisitedState: {
+                funcName: "gpii.firstDiscovery.icon.setVisitedState",
+                args: ["{that}", "{arguments}.0"]
+            }
+        }
+    });
+
+    gpii.firstDiscovery.icon.setActiveState = function (that, isActive) {
+        var activeCss = that.options.styles.active,
+            showCss = that.options.styles.show,
+            activeIndicator = that.locate("activeIndicator"),
+            action = isActive ? "addClass" : "removeClass";
+
+        that.container[action](activeCss);
+        activeIndicator[action](showCss);
```

Comment:
you can use jQuery's `toggleClass` method for this. Then you can just pass in the class and the isActive state.